lib/services/github_issues.rb

```diff
@@ -144,6 +151,17 @@ def receive_webhook
       end
     end
   end
+  
+  def mapped_status(issue)
+    data.status_mapping.to_h[issue.state]
+  end
+  
+  def diff_status_change(resource, issue, diff)
+    if status = mapped_status(issue) and status != resource.workflow_status.id
```

Comment:
Any particular reason to use `and` rather than `&&`?

Reply:
Yep, and it actually baffled me during development at first before I changed it, it's an order of operations thing.

The `and` has really weak precedence and breaks the statement thusly:

`status = mapped_status(issue) and status != resource.workflow_status.id`
`(status = mapped_status(issue)) and (status != resource.workflow_status.id)`

The `&&` construct has higher precedence than assignment, though, so rudely butts into the affair:

`status = mapped_status(issue) && status != resource.workflow_status.id`
`status = (mapped_status(issue) && status != resource.workflow_status.id)`

When I was using the `status` variable it kept coming up `true`, til I realized what was going on.